src/parquet/encodings/plain-encoding.h

```diff
@@ -52,6 +52,19 @@ class PlainDecoder : public Decoder {
     return GetValues(buffer, max_values, sizeof(int64_t));
   }
 
+  virtual int GetInt96(Int96* buffer, int max_values) {
+    max_values = std::min(max_values, num_values_);
+    int size = 3*sizeof(uint32_t);//Struct Int96 could be padded
```

Comment:
I'll let @nongli weigh in; it might be worth doing some performance experiments. The main benefit in a template context is being able to allocate only 12 bytes per buffered value and do pointer arithmetic on `Int96*`. If you allocate `n * sizeof(Int96)` you can still do pointer arithmetic but you are wasting space. 

I've been hearing nowadays that on modern processors, the performance penalty for unaligned memory access is not what it used to be. 

Probably we're doing to want to add a defines that turns on/off struct packing and use `sizeof(Int96)` for memory allocation and assume that pointer arithmetic and dereferencing on works as expected.